### PR TITLE
Adding samples for data save and load from text and binary files

### LINKED ISSUE #3661: Better docs/samples for Data.SaveAsText & Data.SaveAsBinary
I do classification of MS Office document, in my case `feature generation` is relatively slow process, that why I want to do in once and save my data set to file like `data.tsv` and then experiment with different models and parameters.

Let's say we I defined model class
```csharp
private class IrisInputAllFeatures
{
    [ColumnName("Label"), LoadColumn(4)]
    public string IgnoredLabel { get; set; }

    [LoadColumn(0, 3)]
    public float Features { get; set; }
}
```

then I wrote the code that generate / export data

```csharp
var dataSet = new [] {
   new IrisInputAllFeatures() { ... },
   new IrisInputAllFeatures() { ... } 
}
```

the next thing that I expect is to be able to save all my data in file in format supported by `LoadFromTextFile` method (to be able reuse `IrisInputAllFeatures` class in both places),  like

```csharp
Data.SaveToTextFile("DataSet.tsv", dataSet);
```

Is there a way to save data to the file based on `ColumnName`, `LoadColumn` attributes?

### PR DESCRIPTION
**Pull request by najeeb-kazmi**
Fixes #3661

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of training data points.
+            IEnumerable<DataPoint> dataPoints = GenerateRandomDataPoints(10);
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >GenerateRandomDataPoints [](start = 48, length = 24)

no need to use GenerateRandomDataPoints function when training is not needed. you can just define 5 data points inline and keep the sample file shorter.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            IDataView data = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Inspect the data before saving to a binary file.
+            PrintPreviewRows(dataPoints);
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >PrintPreviewRows [](start = 12, length = 16)

if you define the data inline, you don't need the preview. double simplification :)

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            // 0, 0.2060332
+            // 1, 0.5588848
+            // 0, 0.9060271
+            // 1, 0.4421779
+            // 0, 0.9775497
+            // 1, 0.2737045
+
+            // Create a FileStream object and write the IDataView to it as a binary IDV file. 
+            using (FileStream stream = new FileStream("data.tsv", FileMode.Create))
+            {
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >{ [](start = 12, length = 1)

you can drop {} for single statement block (the API samples motto is, keep it as short an to-the-point as possible)

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                mlContext.Data.SaveAsText(data, stream);
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >SaveAsText [](start = 31, length = 10)

please add this file to the xml doc of SaveAsText as example

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            }
+
+            // Create an IDataView object by loading the binary IDV file.
+            IDataView loadedData = mlContext.Data.LoadFromTextFile("data.tsv");
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >LoadFromTextFile [](start = 50, length = 16)

ditto. add to xml.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+            public float Features { get; set; }
+        }
+
+        // Print helper.
+        private static void PrintPreviewRows(IEnumerable<DataPoint> data)
+        {
+            Console.WriteLine($"The rows in the data.");
+            foreach (var row in data)
+                Console.WriteLine($"{row.Label}, {row.Features}");
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >Console.WriteLine($"{row.Label}, {row.Features}"); [](start = 16, length = 50)

you can move this to ToString() definition for DataPoint class and get rid of PrintPreviewRows function altogether. You only need to preview once you read back the data. you could just do the following inside Example() w/o any extra function:
foreach d in data
    WriteLine(d);

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromBinary.cs [hunk 7]
```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.ML;
+
+namespace Samples.Dynamic
+{
+    public static class SaveAndLoadFromBinary
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromBinary.cs]: >SaveAndLoadFromBinary [](start = 24, length = 21)

ditto all the comments in the other file.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromBinary.cs [hunk 8]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.ML;
+
+namespace Microsoft.ML.Samples.Dynamic
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromBinary.cs]: >Microsoft.ML. [](start = 10, length = 13)

remove it.

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+            IDataView data = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Create a FileStream object and write the IDataView to it as a binary IDV file. 
+            using (FileStream stream = new FileStream("data.tsv", FileMode.Create))
+                mlContext.Data.SaveAsText(data, stream);
+
+            // Create an IDataView object by loading the binary IDV file.
+            IDataView loadedData = mlContext.Data.LoadFromTextFile("data.tsv");
+
+            // Inspect the data that is loaded from the previously saved binary file.
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >binary [](start = 73, length = 6)

text

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            };
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            IDataView data = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Create a FileStream object and write the IDataView to it as a binary IDV file. 
+            using (FileStream stream = new FileStream("data.tsv", FileMode.Create))
+                mlContext.Data.SaveAsText(data, stream);
+
+            // Create an IDataView object by loading the binary IDV file.
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >binary [](start = 57, length = 6)

text???

File: docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+                new DataPoint(){ Label = 0, Features = 5},
+                new DataPoint(){ Label = 0, Features = 6},
+                new DataPoint(){ Label = 1, Features = 8},
+                new DataPoint(){ Label = 1, Features = 9},
+            };
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            IDataView data = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Create a FileStream object and write the IDataView to it as a binary IDV file.
```
Comment (zeahmed) [inline docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: >binary [](start = 77, length = 6)

text???

Comment (najeeb-kazmi) [reply docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/SaveAndLoadFromText.cs]: Nice catches @zeahmed. Fixed!

**Pull request merged**